images/prometheus-cloudwatch-exporter/tests/main.tf

```
@@ -0,0 +1,39 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_string" "ref" {
+  input = var.digest
+}
+
+data "oci_exec_test" "cloudwatch-runs" {
+  digest      = var.digest
+  script      = "./cloudwatch-runs.sh"
+  working_dir = path.module
+}
+
+resource "random_pet" "suffix" {}
+
+resource "helm_release" "test" {
+  name       = "cloudwatch-exporter-${random_pet.suffix.id}"
+  repository = "https://prometheus-community.github.io/helm-charts"
+  chart      = "prometheus-cloudwatch-exporter"
+
+  namespace        = "prometheus-cloudwatch-exporter"
+  create_namespace = true
+
+  set {
+    name  = "image.repository"
+    value = data.oci_string.ref.registry_repo
+  }
+  set {
+    name  = "image.tag"
+    value = format("latest@%s", data.oci_string.ref.digest)
```

I think it was me who did this some time back but I also think this can be changed to
```suggestion
    value = data.oci_string.ref.pseudo_tag
```

---

Done!